tensorflow/lite/micro/tools/gen_micro_mutable_op_resolver/generate_micro_mutable_op_resolver_from_model_test.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import os
+import shutil
+import numpy as np
+import tensorflow as tf
+
+from absl import app
+from mako import template
+from tflite_micro.tensorflow.lite.micro.integration_tests import \
+    generate_per_layer_tests as integration_tests
+
+TEMPLATE_DIR = os.path.join(os.path.dirname(__file__), 'templates')
+TEMPLATE_DIR = os.path.abspath(TEMPLATE_DIR)
+
+# FLAGS are reused from integration_tests. Note that final output will be in
+# FLAGS.output_dir/<base name of model>. Where <base name of model> will come
+# from FLAGS.input_tflite_file.
+integration_tests.flags.DEFINE_integer('arena_size', 1024 * 136,
+                                       'Size of arena')
+integration_tests.flags.DEFINE_boolean('verify_output', False,
+                                       'Verify output or just run model.')
+
+
+class MicroMutableOpTestGenerator(integration_tests.TestDataGenerator):
```

Comment:
Sorry, this is an incomplete review, but I like to comment on reusing integration_test.TestDataGenerator:
1.  That class has some built in limitations: it only supports a very small subset of OP as the very first OP. 
2.  It is probably best to extract TestDataGenerator out to its separate py file (generate_test_for_model on top of my head??) to reflect the class is to generate an input and an golden output for a model using tflite interpreter. And then generate_per_layer_tests calls from generate_test_for_model. generate_per_layer_tests is mainly for per layer test, and can assume single layer in there. 

++@advaitjain  WDYT?

Reply:
Sounds good to me.

Reply:
Done